ext/mysqlnd/mysqlnd_ps.c

```diff
@@ -761,15 +761,12 @@ mysqlnd_stmt_fetch_row_buffered(MYSQLND_RES * result, void * param, const unsign
 				}
 
 				for (i = 0; i < result->field_count; i++) {
-					zval *result = &stmt->result_bind[i].zv;
-
-					ZVAL_DEREF(result);
-					/* Clean what we copied last time */
-					zval_ptr_dtor(result);
 					/* copy the type */
-					if (stmt->result_bind[i].bound == TRUE) {
+					zval *resultzv = &stmt->result_bind[i].zv;
+					if (stmt->result_bind[i].bound == TRUE && !Z_ISUNDEF_P(resultzv)) {
 						DBG_INF_FMT("i=%u type=%u", i, Z_TYPE(current_row[i]));
-						ZVAL_COPY(result, &current_row[i]);
+						Z_TRY_ADDREF(current_row[i]);
+						zend_try_assign_ex(resultzv, &current_row[i], 0);
```

Comment:
This might modify `current_row[i]` in-place, which is I think not okay for buffered queries.

Reply:
I guess this is not true in the current implementation, which uses a temporary zval. But per review comments we may want to change it, so I'm pointing it out here. I think it makes sense to have a `ZEND_TRY_ASSIGN_COPY` for this case, which is unambiguous.

Reply:
Correct, also refcount is increased so it anyway should trigger copy-on-write, I guess?
(Or, when we go ahead changing the ZEND_TRY_ASSIGN(), it must probably do a check for RC=1.)